// igdb-api-node/lib/perform-request.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _promise = require('babel-runtime/core-js/promise');

var _promise2 = _interopRequireDefault(_promise);

var _configuration = require('../configuration');

var _configuration2 = _interopRequireDefault(_configuration);

var _request = require('request');

var _request2 = _interopRequireDefault(_request);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = function (url, apiKey) {
    return new _promise2.default(function (resolve, reject) {
        var options = {
            headers: {
                Accept: 'application/json'
            },
            url: url
        };

        if (!_configuration2.default.api.keyHeader) {
            _configuration2.default.api.keyHeader = 'X-Mashape-Key';
        }

        options.headers[_configuration2.default.api.keyHeader] = apiKey;

        (0, _request2.default)(options, function (error, response, body) {
            if (!error) {
                error = response.statusCode === 200 ? null : new Error('HTTP Status ' + response.statusCode + ' - ' + options.url);
            }

            if (error) {
                reject(error);
                return;
            }

            try {
                resolve({
                    body: JSON.parse(body),
                    headers: response.headers,
                    scrollCount: response.headers['X-Count'],
                    scrollUrl: response.headers['X-Next-Page'],
                    url: options.url
                });
            } catch (error) {
                reject(error);
            }
        });
    });
};
